### PR TITLE
comment-popup/svg-connectors/refactoring

### PR DESCRIPTION
**Pull request by munkk**

### CODE REVIEW

**Comment by etokareva** [comment]
Why have you closed your previous PR with my comments?

File: UI/src/app/components/pages/mapping/mapping.component.html [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
-    </app-table>
-
-    <div class="hint-title">
+    
+    <app-area 
+        area='source'>
+        <app-panel *ngFor="let item of (data$ | async).source"
+            [title]="item.table_name">
+            <app-panel-table
+                area='source'
```
Comment (etokareva) [inline UI/src/app/components/pages/mapping/mapping.component.html]: use the same quoting style everywhere.

```
 tableName="item.table_name"
[displayedColumns]="['column_indicator', 'column_name', 'column_type', 'values', 'comments']"
[columns]="item.column_list"
```

File: UI/src/app/components/pages/mapping/mapping.component.ts [hunk 2]
```diff
@@ -14,17 +17,29 @@ export class MappingComponent implements OnInit {
...
+    private httpClient: HttpClient
+  ) {
+    this.common$ = store.pipe(select('common'));
+    this.data$ = store.pipe(select('data'));
   }
 
   ngOnInit() {
-    this.store.dispatch(new Collapsed());
-    this.store.dispatch(new dataActions.FetchData())
+    // this.httpClient.get<any>(`http://127.0.0.1:5000/load_report?path=C:/Projects/CDM/CDMSouffleur/UI/mdcr.xlsx`)
```
Comment (etokareva) [inline UI/src/app/components/pages/mapping/mapping.component.ts]: don't commit to PR commented unworking code (also this part of changes doesn't belong to the refactoring )

File: UI/src/app/components/panel/panel.component.html [hunk 3]
```diff
@@ -1,4 +1,5 @@
-<mat-expansion-panel class="expansion-panel" (click)="onClickMe()">
+<mat-expansion-panel
```
Comment (etokareva) [inline UI/src/app/components/panel/panel.component.html]: no need in additional class. You have the only one mat-expansion panel in each component. Style this tag

File: UI/src/app/directives/draggable.directive.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { DragService } from 'src/app/services/drag.service';
+import { DrawService } from 'src/app/services/draw.service';
+import { elementFromCoords } from 'src/app/utility/kit';
+
+@Directive({
+  selector: '[appDraggable]'
+})
+export class DraggableDirective {
+  @Input('drag-data') data: any;
+  @Input('area') area: string;
```
Comment (etokareva) [inline UI/src/app/directives/draggable.directive.ts]: No need in realiasing
```@Input() area: string;```

File: UI/src/app/services/comments.service.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import { IComment } from 'src/app/models/comment';
+
+@Injectable()
+export class CommentsService {
+  comments = {
+    source: {},
+    target: {}
+  }
+
+  constructor(
```
Comment (etokareva) [inline UI/src/app/services/comments.service.ts]: smth with formatting

File: UI/src/app/store/actions/common.actions.ts [hunk 6]
```diff
@@ -12,8 +12,12 @@ export class Collapsed implements Action {
 
 export class Expanded implements Action {
   readonly type = ActionTypes.Expanded;
+
+  constructor(public payload: any) {}
```
Comment (etokareva) [inline UI/src/app/store/actions/common.actions.ts]: do we need explicit ```public`` word there? Constructors methods ususally are public by default

File: UI/src/app/store/effects/data.effect.ts [hunk 7]
```diff
@@ -23,7 +22,7 @@ export class DataEffect {
     ),
     mergeMap((action: dataActions.FetchData) =>
       this.dataService.retrieveData().pipe(
-        map((data: Data) => 
+        map((data: any) =>
```
Comment (etokareva) [inline UI/src/app/store/effects/data.effect.ts]: if you don't use typechecking, then no need in explicit ```any```

File: UI/src/app/store/reducers/comments.reducer.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    target: Area;
+}
+
+export const initialState: State = {
+  source: {},
+  target: {}
+}
+
+export function commentsReducer(state = initialState, action: CommentsAction) {
+    switch (action.type) {
```
Comment (etokareva) [inline UI/src/app/store/reducers/comments.reducer.ts]: why do we use switch there if we have just one option?

File: UI/src/app/store/reducers/common.reducer.ts [hunk 9]
```diff
@@ -1,32 +1,30 @@
...
 }
 
-export function commonReducer(state = initialState, action: Action) {
+export function commonReducer(state = initialState, action: CommonAction) {
     switch (action.type) {
         case ActionTypes.Collapsed: {
             return {
-                ...state,
-                hint: 'Expand tables to make links'
+                ...state
```
Comment (etokareva) [inline UI/src/app/store/reducers/common.reducer.ts]: all cases have the same return. 
```case ActionTypes.Collapsed:
   case ActionTypes.Expanded:
   case ActionTypes.Linked: {
      // common instructions
   }
```

File: UI/src/app/app.component.html [hunk 10]
```diff
@@ -8,6 +8,7 @@
 </mat-toolbar>
 
 <mat-sidenav-container class="app-sidenav-container">
+
```
Comment (etokareva) [inline UI/src/app/app.component.html]: no need in additional css classes.

File: UI/src/app/components/area/area.component.html [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<h4 class="title">
```
Comment (etokareva) [inline UI/src/app/components/area/area.component.html]: Component CSS styles are encapsulated into the component's view and don't affect the rest of the application if ```encapsulation ``` hasn't been set in the special way. So there is no need to create a bunch of multiple css classes to style components. Ge4t rid of unnecessary classes in your code to make it cleaner

File: UI/src/app/components/dialog/dialog.component.ts [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+  ngAfterViewInit() {
+    this.listener = this.renderer.listen(this.overlay.backdropElement, 'click', () => this.close());
+  }
+
+  loadTemplate(comment: IComment): TemplateRef<any> {
+    if (this.editedComment && this.editedComment.id === comment.id) {
+      return this.editTemplate;
+    } else {
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.ts]: no need in else. Just 
```ts
   return this.readOnlyTemplate;
```
otherwise

File: UI/src/app/components/panel/panel-table/panel-table.component.html [hunk 13]
```diff
@@ -1,25 +1,54 @@
...
-        <td mat-cell *matCellDef="let element">  </td>
+        <td mat-cell *matCellDef="let column">  </td>
     </ng-container>
 
-    <ng-container matColumnDef="symbol">
+    <ng-container matColumnDef="comments">
         <th mat-header-cell *matHeaderCellDef> Comments </th>
-        <td mat-cell *matCellDef="let element"> </td>
+        <td mat-cell *matCellDef="let column">
+            <div #anchor style="width: 40px">
```
Comment (etokareva) [inline UI/src/app/components/panel/panel-table/panel-table.component.html]: do we need to have inline styles? It's not a good practice

**Comment by etokareva** [comment]
> Why have you closed your previous PR with my comments?

Please, work with your opened PRs without cancellation. If you don't know how to work with existing pull requests - you just need to push your changes again to the same branch in which you've opened pull request

File: UI/src/app/components/panel/panel-table/panel-table.component.scss [hunk 14]
```diff
@@ -3,13 +3,42 @@
...
+.caption-item {
+    color: $blue;
+}
+
+.icon-comment {
+    font-size: 14px;
 
-    tbody tr {
-        border-bottom: 1px solid #000;
+    &-blue {
```
Comment (etokareva) [inline UI/src/app/components/panel/panel-table/panel-table.component.scss]: ```icon-comment-blue``` is not informative name. ```icon-comment-active``` would be better

Comment (etokareva) [reply UI/src/app/components/panel/panel-table/panel-table.component.scss]: Or even just ```active``` would be enough. And style it using ```css
   mat-icon.active {
   }
```

File: UI/src/app/components/dialog/dialog.component.ts [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+  close() {
+    this.overlay.detach();
+  }
+
+  isDisabled() {
+    return !!this.editedComment;
+  }
+
+  ngOnDestroy() {
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.ts]: hooks should go right after the constructor

File: UI/src/app/models/comment.ts [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+  newValue(value: string): void;
+}
+
+export class Comment {
+  id: number;
+  date: Date;
+  hasBeenEdited: boolean;
+
+  constructor(
+    public text: string
```
Comment (etokareva) [inline UI/src/app/models/comment.ts]: Do we need explicit ```public``` ?

Comment (munkk) [reply UI/src/app/models/comment.ts]: otherwise how would we create a property "text" here?

Comment (etokareva) [reply UI/src/app/models/comment.ts]: The same way as ```id```

File: UI/src/app/components/dialog/dialog.component.html [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+
+  <input [(ngModel)]="value" [disabled]="isDisabled()" placeholder="Input comment" class="input input-add" (keyup.enter)="add()">
+
+  <button mat-button (click)="add()" [disabled]="isDisabled()" class="button-add" disableRipple="true">
+    Add
+  </button>
+</mat-card>
+
+<ng-template #readOnlyTemplate let-comment>
+  <mat-list-item class="list-item-read">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: rename ```list-item-read```,  ```list-item-edit``` to simple ```read```, ```edit```
and use ```scss
   mat-list-item {
      &.read {
         //styles
      }
   }
```

File: UI/src/app/components/dialog/dialog.component.html [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+    <div class="comment-read">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: ```comment```

File: UI/src/app/components/dialog/dialog.component.html [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+      <div class="comment-read-head">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: ```head```

File: UI/src/app/components/dialog/dialog.component.html [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+        <span class="comment-read-date">
+          {{comment.date | date:'MMM d'}}
+          <ng-container *ngIf="comment.hasBeenEdited">
+            <span>(edited)</span>
+          </ng-container>
+        </span>
+        <div class="comment-read-head-buttons">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: You have nesting, you use preprocessor SCSS. There is no sense to make so long names for css classes

File: UI/src/app/components/dialog/dialog.component.html [hunk 21]
```diff
@@ -1,45 +1,42 @@
...
   </button>
 </mat-card>
 
 <ng-template #readOnlyTemplate let-comment>
-  <mat-list-item class="list-item-read">
-    <div class="comment-read">
-      <div class="comment-read-head">
-        <span class="comment-read-date">
+  <mat-list-item>
+    <div class="read">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: now markup is cleaner, and we can use html5 semantic tags.
I suggest refactor this to 
```html
   <section class="read"> //section, header, main - semantic html5 tags
      <header>
         <span class="date">{{comment.date | date:'MMM d'}}
         <span  *ngIf="comment.hasBeenEdited">(edited)</span> //no need in additional ng-container for ng-if
      </header>
   </section>
```

File: UI/src/app/components/dialog/dialog.component.html [hunk 22]
```diff
@@ -1,45 +1,42 @@
...
+          <button mat-icon-button disableRipple='true' (click)='delete(comment)'>
             <mat-icon>
               delete
             </mat-icon>
           </button>
         </div>
       </div>
-      <div class="comment-read-body">
-        <p class="comment-read-text">
+      <div class="body">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: instead of this use ```main```

File: UI/src/app/components/dialog/dialog.component.html [hunk 23]
```diff
@@ -1,45 +1,42 @@
...
+        <p class="text">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: no need in this tag and class there. just style ```main```

File: UI/src/app/components/dialog/dialog.component.html [hunk 24]
```diff
@@ -48,37 +45,36 @@
 </ng-template>
 
 <ng-template #editTemplate let-comment>
-  <mat-list-item class="list-item-edit">
-    <div class="comment-edit">
-      <div class="comment-edit-head">
-        <span class="comment-edit-date">
+  <mat-list-item>
+    <div class="edit">
```
Comment (etokareva) [inline UI/src/app/components/dialog/dialog.component.html]: make same semantic changes for this part of markup

File: UI/src/app/components/panel/panel-table/panel-table.component.html [hunk 25]
```diff
@@ -1,54 +1,54 @@
...
+        <td mat-cell *matCellDef='let column'>  </td>
     </ng-container>
 
-    <ng-container matColumnDef="comments">
+    <ng-container matColumnDef='comments'>
         <th mat-header-cell *matHeaderCellDef> Comments </th>
-        <td mat-cell *matCellDef="let column">
-            <div #anchor style="width: 40px">
+        <td mat-cell *matCellDef='let column'>
+            <div #anchor>
```
Comment (etokareva) [inline UI/src/app/components/panel/panel-table/panel-table.component.html]: do we need div there?

File: UI/src/app/components/panel/panel.component.html [hunk 26]
```diff
@@ -1,16 +1,13 @@
...
-    
-    <ng-content></ng-content>
-   
-  </mat-expansion-panel>
\ No newline at end of file
+<mat-expansion-panel>
+  <mat-expansion-panel-header>
+    <mat-panel-title>
+      <mat-icon>list</mat-icon>
+      <div>
```
Comment (etokareva) [inline UI/src/app/components/panel/panel.component.html]: do we need div and p there?

File: UI/src/app/components/panel/panel-table/panel-table.component.html (hunk 25 above)
Comment (munkk) [reply UI/src/app/components/panel/panel-table/panel-table.component.html]: yes, otherwise popup won't be positioned properly
